src/aof.c

```diff
@@ -598,40 +596,36 @@ void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv, int a
         cmd->proc == expireatCommand) {
         /* Translate EXPIRE/PEXPIRE/EXPIREAT into PEXPIREAT */
         buf = catAppendOnlyExpireAtCommand(buf,cmd,argv[1],argv[2]);
-    } else if (cmd->proc == setexCommand || cmd->proc == psetexCommand) {
-        /* Translate SETEX/PSETEX to SET and PEXPIREAT */
-        tmpargv[0] = createStringObject("SET",3);
-        tmpargv[1] = argv[1];
-        tmpargv[2] = argv[3];
-        buf = catAppendOnlyGenericCommand(buf,3,tmpargv);
-        decrRefCount(tmpargv[0]);
-        buf = catAppendOnlyExpireAtCommand(buf,cmd,argv[1],argv[2]);
     } else if (cmd->proc == setCommand && argc > 3) {
-        int i;
-        robj *exarg = NULL, *pxarg = NULL;
-        for (i = 3; i < argc; i ++) {
-            if (!strcasecmp(argv[i]->ptr, "ex")) exarg = argv[i+1];
-            if (!strcasecmp(argv[i]->ptr, "px")) pxarg = argv[i+1];
+        robj *pxarg = NULL;
+
+        if (!strcasecmp(argv[3]->ptr, "px")) {
```

Comment:
this `if` heavily relies on the fact the order is known since the command vector is re-written in the command, and we know the index of `px`. i think that's bad.
if we keep it this way, i think a comment is needed here to explain that, and also comment in setGenericCommand next to the argument rewriting, to mention that we have code here that relies on the order.

Reply:
I added comment about the index, I tried to add the `for` loop but it had some issues when comparing the `millisecond` object with `strcasecmp`